drivers/ethernet/eth_enc28j60.c

```diff
@@ -23,88 +23,153 @@
 
 #define D10D24S 11
 
+#ifndef CONFIG_SPI_LEGACY_API
+static struct spi_cs_control enc_spi_cs = {
+	.gpio_pin = 0,
+	.delay = 0
+};
+#endif
```

Comment:
if we are moving this driver to new API, why are we checking for legacy?

Reply:
The patch makes bot SPI API supported at built time.

Reply:
@nashif the driver tests are still using platforms not supporting the new SPI interface, at the moment supporting both API's gives us a higher test coverage

Reply:
> The patch makes both SPI API supported, choice being made at built time.

we should only support one API, and if a platform still support the old API, it will never be moved if we keep this running in this mode and we will end up with 2 APIs for a long time.

I prefer to have this support 1 single API and accelerate the move to the new API with a deadline.

Reply:
@nashif no problem, I agree with that, but there is still the open question, how to handle the tests then?